### DISCUSSION TITLE
[Bug]: Using dataview query with Use All Fields causes an Error

### DISCUSSION THREAD

**Discussion by Bluemoondragon07** [RafaelGB/obsidian-db-folder#878] [Q&A]
### What happened?

Whenever I use a dataview query, such as `from "Articles" where file.name != "Overview"`, and press Use All Fields, the fields will load but it will fail to direct the database at the query and instead load all the files in the current folder. When I go back to validate the query, which was fine before, it says it is not valid. 

Use All Fields works just fine with folders and tags. I also tested the same query before pressing Use All Fields. It loaded in fine then.

### What platform were you using?

Desktop

### Version of the plugin

3.3.0

### Relevant log output

```shell
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'forEach')
    at eval (plugin:dbfolder:118:6542)
    at a (plugin:dbfolder:109:304393)
    at h0.eval (plugin:dbfolder:118:6449)
    at Generator.next (<anonymous>)
    at o (plugin:dbfolder:1:748)
app.js:1 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'top')
    at t.coordsAtPos (app.js:1:1316723)
    at t.updatePosition (app.js:1:1666908)
    at t.showSuggestions (app.js:1:1666606)
    at t.showSuggestions (app.js:1:1668824)
    at t.<anonymous> (app.js:1:1666389)
    at app.js:1:235505
    at Object.next (app.js:1:235610)
    at a (app.js:1:234349)
2plugin:dbfolder:118 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'tags')
    at X0.checkIfFileIsInSource (plugin:dbfolder:118:26887)
    at X0.eval (plugin:dbfolder:118:25714)
    at Generator.next (<anonymous>)
    at eval (plugin:dbfolder:1:938)
    at new Promise (<anonymous>)
    at a (plugin:dbfolder:1:697)
    at Object.n.actions.dataviewUpdater (plugin:dbfolder:118:25644)
    at eval (plugin:dbfolder:109:249809)
    at Generator.next (<anonymous>)
    at eval (plugin:dbfolder:1:938)
images.unsplash.com/reserve/d1Ntvq9mSVmV0RcnWN1Y_23rd%2520Studios%2520Photography%2520Boulder%2520Colorado.jpg?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwzNjAwOTd8MHwxfHNlYXJjaHwyOXx8bmF0dXJlJTIwYXJ0fGVufDB8MHx8fDE2NzUyNzMwNjc&ixlib=rb-4.0.3&q=80&w=1080:1          Failed to load resource: the server responded with a status of 404 ()
```

### Are you using the latest version of the plugin in your Obsidian vault?

- [X] I have verified that I am on the latest version

### Are you check if there is a similar issue?

- [X] I have verified that there are not similar issues

**Comment by RafaelGB** [comment]
I will investigate the cause, thanks for the report =)

**Comment by RafaelGB** [comment]
I tested with multiple cases and I can not reproduce the bug. Could you share some files as example?

Thanks in advance

**Comment by Bluemoondragon07** [comment]
[DB Folder Testing Vault.zip](https://github.com/RafaelGB/obsidian-db-folder/files/10716809/DB.Folder.Testing.Vault.zip)

In this vault, the **Locations Database** is the folder having the issue. The first thing you may notice is that it contains all the fields, but the files are missing. The dataview query has already been typed into this database. If you test this query first thing, it will come up as invalid.

However, once you click **Insert All Fields**, it will say it has an error, but it will successfully load the dataview query. The fields, however, will not be loaded. When you return to test the query again, it will come up as perfectly valid.

Click **Insert All Fields** again, and it will return to the first state.

**Comment by RafaelGB** [comment]
Good morning @Bluemoondragon07 , I found time to study your case. The problem is given by the identifier you use for `Map Location`. In Identificaders, the use of spaces or the Dataview fields will not be allowed.

Due to a function of how the search engine using as a source is not possible today. If Dataview's API changes with the new Datacore project we will be happy to include it
